gui/wxpython/modules/import_export.py

```diff
@@ -470,8 +470,31 @@ def OnRun(self, event):
             if self.dsnInput.GetType() == "dir":
                 idsn = os.path.join(dsn, layer)
             elif self.dsnInput.GetType() == "db":
+                idsn = dsn
                 if "PG:" in dsn:
                     idsn = f"{dsn} table={layer}"
+                elif os.path.exists(idsn):
+                    import subprocess
+
+                    dataset_info = subprocess.run(
+                        ["gdalinfo", "-json", dsn],
+                        capture_output=True,
+                    )
+                    if dataset_info.stderr:
+                        GError(
+                            parent=self,
+                            message=grass.decode(dataset_info.stderr),
+                        )
+                        return
+                    if dataset_info.stdout:
+                        import json
+
+                        dataset_info = json.loads(
+                            grass.decode(dataset_info.stdout),
+                        )
+                        # Rasterlite DB
+                        if "Rasterlite" in dataset_info["driverShortName"]:
```

Comment:
Wouldn't `self.dsnInput.GetFormat` be enough here instead of calling gdalinfo?

Reply:
> Wouldn't `self.dsnInput.GetFormat` be enough here instead of calling gdalinfo?

`self.dsnInput.GetFormat()` return empty string.

Reply:
The solution would be to use the Python Osgeo GDAL package again to get the driver name https://github.com/OSGeo/grass/pull/2513/commits/0f2fccaa1efe22293793f4d22bfd787757fd980f.

Reply:
> > Wouldn't `self.dsnInput.GetFormat` be enough here instead of calling gdalinfo?
> 
> `self.dsnInput.GetFormat()` return empty string.

Any idea why?

Reply:
> > > Wouldn't `self.dsnInput.GetFormat` be enough here instead of calling gdalinfo?
> > 
> > 
> > `self.dsnInput.GetFormat()` return empty string.
> 
> Any idea why?

`self.dsnInput.GetFormat()` return format in the case when you try import raster from Directory radio button widget and you choose format from the Format combo box widget.


<img src="https://github.com/OSGeo/grass/assets/50632337/bc05d2b5-6670-4f78-86b6-56d4cefdea17" width=450>